purchases/src/testDefaults/kotlin/com/revenuecat/purchases/PurchasesTest.kt

```diff
@@ -1835,14 +1963,136 @@ internal class PurchasesTest : BasePurchasesTest() {
+
+        assertThat(capturedUserCancelled).withFailMessage(
+            "Expected userCancelled to be false"
+        ).isNotNull().isFalse()
+    }
+
+    @OptIn(ExperimentalPreviewRevenueCatPurchasesAPI::class)
+    @Test
+    fun `initial purchase with add-ons throws when Store is not PLAY_STORE`() {
+        val purchaseParams = PurchaseParams.Builder(
+            activity = mockActivity,
+            storeProduct = stubStoreProductWithGoogleSubscriptionPurchaseData()
+        )
+            .addOnStoreProducts(
+                addOnStoreProducts = listOf(stubStoreProductWithGoogleSubscriptionPurchaseData(productId = "abc"))
+            )
+            .build()
+
+        for (store in Store.values()) {
+            if (store == Store.PLAY_STORE) { continue }
+            buildPurchases(
+                anonymous = false,
+                apiKeyValidationResult = APIKeyValidator.ValidationResult.SIMULATED_STORE,
+                enableSimulatedStore = true,
+                store = store
+            )
+
+            val latch = CountDownLatch(1)
+            var capturedError: PurchasesError? = null
+            var capturedUserCancelled: Boolean? = null
+
+            purchases.purchase(
+                purchaseParams = purchaseParams,
+                callback = object: PurchaseCallback {
+                    override fun onCompleted(storeTransaction: StoreTransaction, customerInfo: CustomerInfo) {
+                        fail("purchase() should fail with a purchase with add-ons when the store is not the Play Store")
+                        latch.countDown()
+                    }
+
+                    override fun onError(error: PurchasesError, userCancelled: Boolean) {
+                        capturedError = error
+                        capturedUserCancelled = userCancelled
+                        latch.countDown()
+                    }
+                },
+            )
+
+            assertThat(latch.await(3, TimeUnit.SECONDS)).withFailMessage(
+                "Callback was not called within timeout for store: $store"
+            ).isTrue()
+
+            assertThat(capturedError).withFailMessage(
+                "Expected error to be captured for store: $store"
+            ).isNotNull()
+
+            assertThat(capturedError?.code).isEqualTo(PurchasesErrorCode.PurchaseInvalidError)
+            assertThat(capturedError?.message).isEqualTo(PurchasesErrorCode.PurchaseInvalidError.description)
+            assertThat(capturedError?.underlyingErrorMessage).isEqualTo("Making a purchase with add-ons is only supported on" +
+                " the Play Store.")
+
+            assertThat(capturedUserCancelled).withFailMessage(
+                "Expected userCancelled to be false for store: $store"
+            ).isNotNull().isFalse()
+        }
+    }
+
+    @OptIn(ExperimentalPreviewRevenueCatPurchasesAPI::class)
+    @Test
+    fun `product change with add-ons throws when Store is not PLAY_STORE`() {
```

Comment:
The diff here looks a little funny. What's happening here is that I added a copy of the `initial purchase with add-ons throws when Store is not PLAY_STORE` test, but for product changes :)